rfcs/text/0015_bazel.md

```diff
@@ -0,0 +1,308 @@
+
+The `@kbn/pm` package was updated in https://github.com/elastic/kibana/pull/89961 to run the new packages build target, invoked by calling `bazel build //packages:build`, before executing the existing legacy package builds.
+
+The build targets will no longer reside within the package themselves and instead will be within the `bazel/bin` directory. To account for this, any defined dependency will need to be updated to reference the new directory (example: `link:bazel/bin/packages/elastic-datemath`). While also in this transition period, the build will need to copy over the packages from `bazel/bin` into the `node_modules` of the build target.
+
+Example package BUILD.bazel for `packages/elastic-datemath`:
+
+```python
+load("@build_bazel_rules_nodejs//:index.bzl", "pkg_npm")
+load("@build_bazel_rules_nodejs//internal/js_library:js_library.bzl", "js_library")
+
+SRCS = [
+    ".npmignore",
+    "index.js",
+    "index.d.ts",
+    "package.json",
+    "readme",
+    "tsconfig.json",
+]
+
+filegroup(
+    name = "src",
+    srcs = glob(SRCS),
+)
+
+js_library(
+    name = "elastic-datemath",
+    srcs = [ ":src" ],
+    deps = [ "@npm//moment" ],
+    package_name = "@elastic/datemath",
+    visibility = ["//visibility:public"],
+)
+
+alias(
+    name = "build",
+    actual = "elastic-datemath",
+    visibility = ["//visibility:public"],
+)
+```
+
+If the package has unit tests, they will need to be migrated which will be invoked with `bazel test` as described in the Unit Testing section.
+
+
+## Phase II - Docs, Developer Experience
+
+Packages were a likely choice for phase 1 for a few reasons; they aren’t often updated and the developer experience is quite lacking making it easy to maintain parity with. In phase 2, we will bring the developer experience of packages to that which developers are accustomed to with plugins. This means re-builds will be automatic when a change occurs as well as giving time to address any developer experience shortcomings which were not foreseen. During this time we will work on overall Bazel documentation as it pertains to the Kibana repository.
+
+
+## Phase III - Plugins
+
+In this phase, we will be migrating each of the 135 plugins over to being built and unit tested using Bazel. During this time, the legacy systems will stay in place and run in parallel with Bazel. Once all plugins have been migrated, we can decommission the legacy systems.
+
+The `BUILD.bazel` files will look similar to that of packages, there will be a target for `web`, `server`, and `jest`. Just like packages, as the Jest unit tests are migrated, they will need to be removed from the root `jest.config.js` file as described in the Unit Testing section.
+
+Plugins are built in a sandbox, so they will no longer be able to use relative imports from one another. For Typescript, relative imports will be replaced with a path reference to the `bazel/bin`.
```

Comment:
Can we add an example of what this would look like in code? I think how these imports work is an important aspect of how well this tool will fit into our repo. It may also impact how we consider implementing https://github.com/elastic/kibana/issues/71566 and enforce encapsulation between plugins & domains.

Reply:
I agree that we should elaborate on this change and discuss it with a wider audience. A dedicated e-mail to kibana-contributors@ would be great.

Reply:
Started https://github.com/elastic/kibana/issues/94034 to attempt to get consensus on this: https://github.com/elastic/kibana/issues/94034. Will pass it around to a few folks before blasting kibana-contributors.

Reply:
Thanks, @tylersmalley.